src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java

```diff
@@ -41,51 +48,75 @@ public ClaimService(
         JsonMapper jsonMapper,
         IssueDateCalculator issueDateCalculator,
         ResponseDeadlineCalculator responseDeadlineCalculator,
-        EventProducer eventProducer) {
+        EventProducer eventProducer,
+        InterestCalculationService interestCalculationService) {
         this.claimRepository = claimRepository;
         this.userService = userService;
         this.jsonMapper = jsonMapper;
         this.issueDateCalculator = issueDateCalculator;
         this.responseDeadlineCalculator = responseDeadlineCalculator;
         this.eventProducer = eventProducer;
+        this.interestCalculationService = interestCalculationService;
     }
 
     public Claim getClaimById(long claimId) {
-        return claimRepository
-            .getById(claimId)
-            .orElseThrow(() -> new NotFoundException("Claim not found by id " + claimId));
+        return calculateAndPopulateTotalAmount(
+            claimRepository
+                .getById(claimId)
+                .orElseThrow(() -> new NotFoundException("Claim not found by id " + claimId))
+        );
     }
 
     public List<Claim> getClaimBySubmitterId(String submitterId) {
-        return claimRepository.getBySubmitterId(submitterId);
+        return claimRepository.getBySubmitterId(submitterId).stream()
+            .map((claim) -> calculateAndPopulateTotalAmount(claim)).collect(Collectors.toList());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1612)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "claim" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1611)